python/python_helper_sympy.py

```diff
@@ -221,64 +273,70 @@ def evaluate(expr, locals_for_eval={}):
 
     # Convert AST to code and evaluate it with no global expressions and with
     # a whitelist of local expressions
-    locals = {}
-    for key in locals_for_eval:
-        locals = {**locals, **locals_for_eval[key]}
+    locals = dict(chain.from_iterable(
+        cast(SympyMapT, local_expressions).items() for local_expressions in locals_for_eval.values()
+    ))
+
     return eval(compile(root, '<ast>', 'eval'), {'__builtins__': None}, locals)
 
-def convert_string_to_sympy(a, variables, allow_hidden=False, allow_complex=False):
+def convert_string_to_sympy(expr: str, variables: Optional[List[str]], *, allow_hidden: bool=False, allow_complex: bool=False, allow_trig_functions: bool=True) -> sympy.Expr:
```

Comment:
do you need the `bool` annotations here, or does having the default be `False` make the typechecker assume bool? I think typescript does this, IDK about python.

Reply:
You do need the bool. Anything without annotations is automatically treated as an `Any`.

Reply:
that's unfortunate.